amethyst_renderer/src/pass/util.rs

```diff
@@ -249,14 +249,24 @@ pub fn set_vertex_args(
 ) {
     let vertex_args = camera
         .as_ref()
-        .map(|&(ref cam, ref transform)| VertexArgs {
-            proj: cam.proj.into(),
-            view: transform.0.invert().unwrap().into(),
-            model: global.0.into(),
-        }).unwrap_or_else(|| VertexArgs {
-            proj: Matrix4::one().into(),
-            view: Matrix4::one().into(),
-            model: global.0.into(),
+        .map(|&(ref cam, ref transform)| {
+            let proj: [[f32; 4]; 4] = cam.proj.into();
+            let view: [[f32; 4]; 4] = transform.0.try_inverse().unwrap().into();
+            let model: [[f32; 4]; 4] = global.0.into();
```

Comment:
I'm surprised those aren't implicit since we don't have a generic type over Transform & others

Reply:
What do you mean?

Reply:
The compiler should figure the type it by itself normally, considering you are using it the line right under.

Reply:
Ah, that could be done before since that `glsl-layout` has a cgmath feature which will allow cgmatch types but since there isn't such a thing for nalgebra we need to do `Matrix -> [[f32; 4]; 4] -> mat4`.